ivy/functional/ivy/experimental/manipulation.py

```diff
@@ -710,11 +710,19 @@ def _set_pad_area(padded, axis, width_pair, value_pair):
     if width_pair[0] > 0:
         left_slice = _slice_at_axis(slice(None, width_pair[0]), axis)
         padded[left_slice] = value_pair[0]
+        if isinstance(value_pair[0], ivy.Array):
+            padded[left_slice] = value_pair[0].data
+        else:
+            padded[left_slice] = value_pair[0]
     if width_pair[1] > 0:
         right_slice = _slice_at_axis(
             slice(padded.shape[axis] - width_pair[1], None), axis
         )
         padded[right_slice] = value_pair[1]
+        if isinstance(value_pair[1], ivy.Array):
+            padded[right_slice] = value_pair[1].data
+        else:
+            padded[right_slice] = value_pair[1]
```

Comment:
When I was testing the pad function on multiple test cases. sometimes it raises errors like this: 

self = ivy.array([0]), args = (dtype('uint8'),), kwargs = {}

    def __array__(self, *args, **kwargs):
        args, kwargs = args_to_native(*args, **kwargs)
>       return self._data.__array__(*args, dtype=self.dtype, **kwargs)
E       TypeError: __array__() got multiple values for argument 'dtype'

on these two lines:
`padded[right_slice] = value_pair[1]`
`padded[left_slice] = value_pair[0]`

I debugged this many times and tried to find the root cause of this error and check why it is happening. I find out that sometimes the _set_pad_area helper function gets the value_pair as ivy array (not most of the time because most of the time value_pair is just a list) and I don't know why it gets this sometimes but the reason for the error, when it gets ivy array, was that it already has dtype in the *args and it again assigning the dtype using self.dtype which raises an error of "multiple values of dtype arguments". on the other hand, when the value pair is list then it does not have multiple dtype so it works fine,

I think when the value_pair is ivy array then it has two dtypes when we run this line: padded[left_slice] = value_pair[0] or padded[right_slice] = value_pair[1], and it gets confused to add the first dtype or second one. So that's why I only fetch the data part of the ivy array which returns a native array and then we store it into padded and then it gets only one dtype.

Maybe we can find a better solution for this but at the moment this solution works fine and does not fail on these corner cases

Reply:
Thanks for explaining this! Could you give me an example set of inputs that cause this error. I'd like to have a look at it myself and push a fix directly to master. There might be a more elegant solution.

Reply:
Nevermind, I just got the error after running `test_pad` a couple more times.